src/Symfony/Component/Workflow/EventListener/GuardListener.php

```diff
@@ -55,6 +55,10 @@ private function getVariables(GuardEvent $event)
     {
         $token = $this->tokenStorage->getToken();
 
+        if (null === $token) {
+            throw new \Exception('No token is set');
```

Comment:
I don't know why @stloyd's comment is lost ; but he was right.
You should use an exception that lives in the Workflow component. If the existing exception does not match
this case, you will have to create a new one.